IoTBlockly/SimpleWebServer/SimpleWebServer.cs

```diff
@@ -246,90 +248,104 @@ public void Post(string path, RouteCallback callback)
             postRoutes.Add(Tuple.Create<string, RouteCallback>(path, callback));
         }
 
+
         private async void ProcessRequestAsync(StreamSocket socket)
         {
             // very rudimentary web server request processing
-
-            // grab the request
-            StringBuilder request = new StringBuilder();
-            byte[] data = new byte[BufferSize];
-            IBuffer buffer = data.AsBuffer();
-            uint dataRead = BufferSize;
-            using (IInputStream input = socket.InputStream)
+            try
             {
-                while (dataRead == BufferSize)
+                // grab the request
+                StringBuilder request = new StringBuilder();
+                byte[] data = new byte[BufferSize];
+                IBuffer buffer = data.AsBuffer();
+                uint dataRead = BufferSize;
+
+                using (IInputStream input = socket.InputStream)
                 {
-                    await input.ReadAsync(buffer, BufferSize, InputStreamOptions.Partial);
-                    request.Append(Encoding.UTF8.GetString(data, 0, data.Length));
-                    dataRead = buffer.Length;
+                    while (dataRead == BufferSize)
```

Comment:
is there a case when dataRead != BufferSize but dataRead > 0? are we leaving some bytes on the ground?